### PR TITLE
Added Rest APIs to add/remove contributors in a channel

### LINKED ISSUE #132: Add/remove channel contributor API
Users need to be added as contributors to see the content of a private channel.

The `remove` part must not remove administrators (moderators in the reddit terminology)

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
closes #132 

#### What's this PR do?
Adds a REST API to add and remove contributors in a channel

#### How should this be manually tested?
if you disable the `JwtIsStaffOrReadonlyPermission` in the views, you can use the django rest framework UI to perform the requests and you can go to `http://reddit.local/r/<channel_name>/about/contributors/` and verify that the action you perform via rest API correspond to what you see in the UI

### CODE REVIEW

File: channels/api.py [hunk 1]
```diff
@@ -361,3 +366,52 @@ def more_comments(self, comment_fullname, parent_fullname, count, children=None)
...
+
+        """
+        try:
+            user = User.objects.get(username=contributor_name)
+        except User.DoesNotExist:
+            raise NotFound("User {} does not exist".format(contributor_name))
+        channel = self.get_channel(channel_name)
+        if contributor_name in channel.moderator():
+            raise exceptions.RemoveUserException(
+                "User {} is a moderator of the channel{}".format(contributor_name, channel_name))
```
Comment (noisecapella) [inline channels/api.py]: Can you add a space between `channel` and `{}`?

**Comment by noisecapella** [comment]
Functionality looks great

File: channels/urls.py [hunk 2]
```diff
@@ -20,6 +22,16 @@
         name='post-list',
     ),
     url(
+        r'^api/v0/channels/(?P<channel_name>[A-Za-z0-9_]+)/contributors/$',
+        ContributorListView.as_view(),
+        name='contributors-list',
```
Comment (noisecapella) [inline channels/urls.py]: Can you change this to `contributor-list` and the below to `contributor-detail`, for consistency with the other URL names?

File: channels/api_test.py [hunk 3]
```diff
@@ -261,3 +266,56 @@ def test_frontpage(mock_client):
...
+    assert mock_client.subreddit.return_value.contributor.remove.call_count == 0
+
+
+def test_remove_contributor(mock_client):
+    """Test remove contributor"""
+    client_user = UserFactory.create()
+    contributor = UserFactory.create()
+    client = api.Api(client_user)
+    client.remove_contributor(contributor.username, 'foo_channel_name')
+    mock_client.subreddit.return_value.contributor.remove.assert_called_once_with(contributor)
```
Comment (annagav) [inline channels/api_test.py]: this only tests if the method was called, but it does not test if the user was removed from the list of contributors.

Comment (giocalitri) [reply channels/api_test.py]: well... the assumption is that if the library is called and it does not return an error, the user has been removed.

Actually the problem is even more complicated, given that `praw` will not return errors if the user is not a contributor, for example.

**Pull request merged**